### PR TITLE
Fix bugs

### LINKED ISSUE #225: [PE-D] Insufficient negative examples for User Guide
It would be great for the reader to see some of the negative examples of using the application and the explanation of why it was wrong. 

One particular example is the add/edit for reminders, understand the following examples on reminders quite straightforward but would be good to mention it! :)

Does not state that reminders need to be after current date:

Same for role restrictions, only stated the format but not the reminder date restrictions:

Only seen on the application itself which can be quite confusing for the user:

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: whoisjunhong/ped#2

### PR DESCRIPTION
**Pull request by BrendonLau**
1. Added optional fields in message for invalid addCompany command
2. Added optional fields in message for invalid addRole command
3. comparing of CompanyName will be case insensitive
4.  comparing of RoleName will be case insensitive

for point 3 & 4 adding another role name/ company name with existing role/company with a different case in name will not be allowed

5. Remove trailing zeros and only allow positive number for stipend
6. Update stipend error message
7. Update respective testing for each changes

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -249,7 +249,8 @@ The following points explain the format of a command.
 ### Role Restrictions <a id="c-role-restrictions"></a>
 
 * The `ROLE_NAME` should only contain alphanumeric characters, spaces and an optional pair of round brackets.
-* The `REMINDER_DATE` should be in format dd-MM-yyyy HH:mm.
+* The `REMINDER_DATE` should not be in the past and must be a valid date
```
Comment (kerbysoh) [inline docs/UserGuide.md:252]: Nice job making it more clear to users that the reminder should not be in the past.

File: src/main/java/seedu/tinner/logic/commands/AddRoleCommand.java [hunk 2]
```diff
@@ -27,11 +27,11 @@
     public static final String MESSAGE_USAGE = COMMAND_WORD + ": Adds a role to the company listed. "
             + "Parameters: "
             + "companyIndex "
-            + PREFIX_NAME + "ROLE_NAME (TYPE) "
+            + PREFIX_NAME + "ROLE_NAME [(TYPE)] "
             + PREFIX_STATUS + "STATUS "
-            + PREFIX_REMINDER_DATE + "REMINDER_DATE "
-            + PREFIX_DESCRIPTION + "DESCRIPTION "
-            + PREFIX_STIPEND + "STIPEND\n"
+            + PREFIX_REMINDER_DATE + "[REMINDER_DATE] "
```
Comment (kerbysoh) [inline src/main/java/seedu/tinner/logic/commands/AddRoleCommand.java]: I feel that the brackets here should be across both the prefix and the value to be more specific

**Review by kerbysoh — APPROVED** [review]
LGTM other than minor issue pointed out

File: src/main/java/seedu/tinner/logic/commands/AddRoleCommand.java (hunk 2 above)
Comment (BrendonLau) [reply src/main/java/seedu/tinner/logic/commands/AddRoleCommand.java]: Oops! thanks for spotting. Made the changes.

**Pull request merged**